src/scripts/h5p-dictation-sentence.js

```diff
@@ -108,11 +109,46 @@ class Sentence {
     );
 
     // Text input field
-    this.inputField = document.createElement('input');
+    this.inputField = document.createElement('textarea');
+    this.inputField.setAttribute('rows', 1);
     this.inputField.setAttribute('spellcheck', 'false');
     this.inputField.setAttribute('autocorrect', 'off');
     this.inputField.setAttribute('autocapitalize', 'off');
     this.inputField.setAttribute('aria-label', this.params.a11y.enterText);
+
+    // Prevent ENTER key
+    this.inputField.addEventListener('keypress', (e) => {
+      if (e.keyCode === 13 || e.which === 13 || e.key.charCodeAt(0) === 13) {
+        e.preventDefault();
+        return false;
+      }
+    });
+    
+    [ 'change', 'keyup' ].forEach(event => this.inputField.addEventListener(event, () => {
+      // Remove new lines ( in case of pasting, for instance )
+      if (this.inputField.value.indexOf('\n') !== -1 || this.inputField.value.indexOf('\r') !== -1) {
+        this.inputField.value = this.inputField.value.replace(/[\n\r]/g, '');
+      }
+
+      // Autosize
+      if (!this.inputField.style.height || this.inputField.scrollHeight !== parseInt(this.inputField.style.height)) {
+        let heightDelta = parseInt(this.inputField.getAttribute('heightDelta'));
+
+        if (!heightDelta) {
+          this.inputField.setAttribute('heightDelta', this.inputField.offsetHeight - this.inputField.clientHeight );
+          heightDelta = parseInt(this.inputField.getAttribute('heightDelta'));
+        }
+        
+        // Set size to zero to allow shrinking
+        this.inputField.style.height = 0;
+
+        // Set size to it's content size added of delta ( borders )
+        this.inputField.style.height = (this.inputField.scrollHeight + heightDelta) + 'px';
+
+        // Notifies H5P so iframe is resized
+        this.params.callbacks.resize();
+      }
+    }));
     this.inputField.classList.add(Sentence.INPUT_FIELD);
 
     // Restore previous state
```

Comment:
Conclusion: I had a look at that flickering. There's probably nothing that you can do if you use the keyup listener. The browser will have rendered and scrolled the text. I tried setting `scrollTop` to 0, but that wasn't processed in time.

I had a look at the specs of the fields' events. `keyup` doesn't work, as I just mentioned. It seemed that `input` would be a good choice, because it's also fired whenever the value changes. Lucky shot, but it seems to do the trick.

This should do the job nicely without flickering, just one listener instead of three, discards "enter" and only requests a resize from H5P core when the size of the field actually changed:
```
    // Auto resize the input field
    this.inputField.addEventListener('input', () => {
      // Remove line breaks when pasting, etc.
      if (this.inputField.value.indexOf('\n') !== -1 || this.inputField.value.indexOf('\r') !== -1) {
        this.inputField.value = this.inputField.value.replace(/[\n\r]/g, '');
      }

      this.inputField.style.height = 'auto'; // Reset to allow shrinking
      const needsResize = (this.previousScrollHeight !== this.inputField.scrollHeight);

      this.inputField.style.height = `${this.inputField.scrollHeight + this.inputField.offsetHeight - this.inputField.clientHeight}px`;

      if (needsResize) {
        this.previousScrollHeight = this.inputField.scrollHeight;

        // Trigger iframe resize
        this.params.callbacks.resize();
      }
    });
```

Reply:
Cool! Was busy with other things, let's give it a try.

Reply:
Feel free to put that into the pull request.

Reply:
@tdjac0bs Not pushing, but it seems the H5P core team is preparing to update the external content types. I could include the code myself if you don't have the time, but since you were so kind to take care about that feature, I thought you might want to try it out yourself first and have the honor of adding the code.

Reply:
hello @otacke - thanks for heads up.
Will update PR by end of today.